cmd/soroban-cli/src/commands/contract/build.rs

```diff
@@ -295,8 +299,54 @@ impl Cmd {
             return Ok(());
         }
 
-        let mut wasm_bytes = fs::read(target_file_path).map_err(Error::ReadingWasmFile)?;
+        // get existing wasm bytes
+        let wasm_bytes = fs::read(target_file_path).map_err(Error::ReadingWasmFile)?;
+        let existing_meta: Vec<ScMetaEntry> = Spec::new(&wasm_bytes).unwrap().meta;
+
+        let mut module = Module::new();
+
+        for payload in WasmParser::new(0).parse_all(&wasm_bytes) {
+            match payload.unwrap() {
+                Payload::CustomSection(section) => {
+                    if section.name() == META_CUSTOM_SECTION_NAME {
+                        let updated_meta = self.append_new_meta(&existing_meta).unwrap();
+                        let custom = CustomSection {
+                            name: section.name().into(),
+                            data: updated_meta.into(),
+                        };
+                        module.section(&custom);
+                    } else {
+                        let custom = CustomSection {
+                            name: section.name().into(),
+                            data: section.data().into(),
+                        };
+                        module.section(&custom);
+                    }
+                }
+                other => {
+                    // Reconstruct raw section bytes and add them to the new module
+                    if let Some((id, range)) = other.as_section() {
+                        let raw = RawSection {
+                            id,
+                            data: &wasm_bytes[range.start..range.end],
+                        };
+                        module.section(&raw);
+                    }
+                }
+            }
+        }
+
+        let updated_wasm_bytes = module.finish();
 
+        // Deleting .wasm file effectively unlinking it from /release/deps/.wasm preventing from overwrite
+        // See https://github.com/stellar/stellar-cli/issues/1694#issuecomment-2709342205
+        fs::remove_file(target_file_path).map_err(Error::DeletingArtifact)?;
+        fs::write(target_file_path, updated_wasm_bytes).map_err(Error::WritingWasmFile)
+    }
+
+    fn append_new_meta(&self, existing_meta: &[ScMetaEntry]) -> Result<Vec<u8>, Error> {
+        let mut updated_meta = existing_meta.to_owned();
```

Comment:
This line makes a copy of the vec in memory, which is unnecessary because the only use of the vec being passed in is to append. So you can accept a `&mut Vec<ScMetaEntry>` and modify it directly to avoid the copy.